### ISSUE TITLE
Set the hostname for webpage redirections

### ISSUE THREAD

**Issue by antofthy** [codercom/code-server#582]
## Description

I run coder server in a docker swarm.
      code-server ~/html -d ~/code-server
Which connects port 8443 to a virtual web proxy.

For example:   https://xyzzy.example.org:8443/
will be directed the port 8443 on the appropriate container

The response I get back is a redirection to
   https://loclhost:8443/login
Which is of course completely wrong!

How can I tell code-server that the hostname it should do its redirections for is "xyzzy.example.org"  and NOT   "localhost"

the --host argument seems to be to specify the binding host for the port (aka -p ) whcih is why it defaults to 0.0.0.0 

## Related Issues

testing with "curl" from the container I 
There is seems to be a LOT of redirections for the initial login

http://localhost:8443/  ->
https://localhost:8443/ ->
https://localhost:8443/login ->
https://localhost:8443/login/

At that point I get an error
Error: ENOENT: no such file or directory, open '/home/travis/build/codercom/code-server/packages/server/build/web/login/index.html'

**Comment by antofthy** [comment]
Look like the problems went away!   Except the last continues when using 'curl' for testing.

With an actual browser (using  --no-auth and --allow-html)  I now just get a blank screen with the html source showing some java code.

**Comment by Egoscio** [comment]
@antofthy What "virtual web proxy" are you using? I had the same issue, and I found out that in order to properly fix it, you need to make your proxy "transparent". That basically means your proxy has to forward the `Host` header.

**Comment by antofthy** [comment]
Yes, I was able to finally resolve that problem.

What I have been unable to resolve is using the Binary download in separately build docker environment.   I get a blank web page.   I am sure there is something missing in the environment that the binary needs,  Perhaps access to other files from the source image, or some library, but there is no error report, or any 'requirements' outlined as needed for the binary.

On the other hand if I use the docker image they provide as a straight docker run image, it works fine. But I need to be able to run multiple copys of code-server in a docker swarm (seperated multiple-users for a class environment), meaning I need to run it as a 'service' in a swarm environment.   IN taht situation it does not work,   which is why I am trying to create my own docker image.

In summary:  The binary in a swarm works, but gives a blank page, and the pre-build docker image does not work in a swarm.   So I have made progress, but have not succeeded as yet.

-----

Example...
    Start a php:7.3-apache service in a docker swarm.
    Run a BASH command line that docker container, and download, install and binary code-server.
```
  curl -s -L https://github.com/codercom/code-server/releases/latest |
      grep download | grep tar.gz |
      sed '2q; s%.*href="%https://github.com%; s/".*//' |
      xargs curl -s -L -o - | tar zxvf -
    mv code-server*/ code-server
    chmod +x code-server/code-server
    code-server/code-server -d ~/code-server ~/html --no-auth --allow-html -p 8080 &
```
At this point I can connect to the code-server web page on port 8080, via the swarm ingress proxy
(which currently allows http on 80, and 8080, and https on 8443).
But I only see a blank page.

Same if I remove the option`--allow-html -p 8080` and connect via https on port 8443

The blank page is why I theorize the provided binary requires something MORE that is not documented, or provided in the tar file for the binary.

I also see the following error, from the command line connection to the php:7.3-apache docker service
`Error: ENOENT: no such file or directory, open '/home/travis/build/codercom/code-server/packages/server/build/web/index.html'`

Which presumably is the path to the build directory of the provided binary, which is NOT present, in the tar file of the binary, or is effected by any option I have tried.

I have seen a similar error in the provided docker image...
`Error: ENOENT: no such file or directory, open '/src/packages/server/build/web/index.html'`
which does match the build directory used in the 'Dockerfile' (and is still present in the image) that was used to build that docker image.

It is very frustrating.
